tasklist/client/src/Tasks/CollapsiblePanel/index.tsx

```diff
@@ -0,0 +1,230 @@
+    sortBy === undefined
+      ? {filter}
+      : {filter, sortBy};
+  if (filter === 'custom') {
+    const customFilters = getStateLocally('customFilters')?.custom;
+    const customFiltersParams =
+      filter === 'custom' && customFilters !== undefined
+        ? prepareCustomFiltersParams(customFilters, userId)
+        : {};
+    const updatedParams =
+      Object.keys(customFiltersParams).length > 0
+        ? new URLSearchParams({
+            ...convertedParams,
+            ...values,
+            ...customFiltersParams,
+          })
+        : new URLSearchParams({
+            ...convertedParams,
+            ...values,
+          });
+    const paramsToDelete = difference(
+      CUSTOM_FILTERS_PARAMS,
+      Object.keys(customFiltersParams),
+    );
+
+    paramsToDelete.forEach((param) => {
+      updatedParams.delete(param);
+    });
+
+    return updatedParams.toString();
+  }
+
+  const updatedParams = new URLSearchParams({
+    ...convertedParams,
+    ...values,
+  });
+
+  if (filter === 'completed') {
+    updatedParams.set('sortBy', 'completion');
+  }
+
+  CUSTOM_FILTERS_PARAMS.forEach((param) => {
+    updatedParams.delete(param);
+  });
+
+  return updatedParams.toString();
+}
+
+const CollapsiblePanel: React.FC = () => {
+  const [isCollapsed, setIsCollapsed] = useState(true);
+  const [searchParams] = useSearchParams();
+  const customFilters = getStateLocally('customFilters')?.custom;
+  const {data} = useCurrentUser();
+  const userId = data?.userId ?? '';
+
+  if (isCollapsed) {
+    return (
+      <div className={cn(styles.base, styles.collapsedContainer)}>
+        <Button
+          hasIconOnly
+          iconDescription="Expand to show filters"
+          tooltipPosition="right"
+          onClick={() => {
+            setIsCollapsed(false);
+          }}
+          renderIcon={SidePanelOpen}
+          size="md"
+          kind="ghost"
+        />
+        <Button
+          hasIconOnly
+          iconDescription="Custom filter"
+          tooltipPosition="right"
+          onClick={() => {}}
+          renderIcon={Filter}
+          size="md"
+          kind="ghost"
+        />
+      </div>
+    );
+  }
+
+  return (
+    <div className={cn(styles.base, styles.expandedContainer)}>
+      <span className={cn(styles.header, sharedStyles.panelHeader)}>
+        <h1>Filters</h1>
+        <Button
+          hasIconOnly
+          iconDescription="Collapse "
+          tooltipPosition="right"
+          onClick={() => {
+            setIsCollapsed(true);
+          }}
+          renderIcon={SidePanelClose}
+          size="md"
+          kind="ghost"
+        />
+      </span>
+      <SearchParamNavLink
+        to={{
+          search: getNavLinkSearchParam({
+            currentParams: searchParams,
+            filter: 'all-open',
+            userId,
+          }),
+        }}
+        activeParam={{
+          key: 'filter',
+          value: 'all-open',
+        }}
+        isActiveOnEmpty
```

Comment:
What are your thoughts on having a `isActive` prop? This is the only instance that uses `isActiveOnEmpty`.

```
isActive={filter === 'all-open' || filter === null}
```
Just seems like you're passing a lot of data to do this comparison in the Link component rather than here.

Reply:
yep, this is better